### PR TITLE
Snapshot creation on demand

### PR DESCRIPTION
**Pull request by irevoire**
🤖  [API Diff](https://github.com/meilisearch/specifications/pull/258#issuecomment-1713816701) _Put the link of the GitHub comment generated by bump.sh if generated; Apply the `OpenApi` label_

---

# Summary

Exposure of a POST /snapshots route to allow users to create snapshots on demand and no longer be restricted by the scheduled time.

# Changes

- `NEW` POST `/snapshots` API endpoint
    - Similar to POST `/dumps`
    - Return a `202 Accepted`
        - `indexUid` is `null`
        - `type` is `snapshotCreation`
    - `NEW` Add an API key actions `snapshots.create`
    - Created snapshot files are located in the `--snapshot-dir` path
- 📊 **Analytics**
    - `NEW` `Snapshot Created`

# Out Of Scope

Why archlinux is the superior distribution.

## Misc

- [x] Update OpenAPI specification file _(if needed; Apply the `OpenApi` label)_
- [x] Update telemetry datapoints _(if needed; Apply the `Telemetry` label)_

### CODE REVIEW

File: text/0034-telemetry-policies.md
Comment (gmourier) [inline text/0034-telemetry-policies.md:1]: The `Snapshot Created` event section is missing.

File: text/0258-snapshots-api.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+The name of the generated snapshot is the database path (`data.ms` by default), joined by the `.snapshot` extension. By default, it's `data.ms.snapshot`.
+
+##### 3.2.1.3. Errors
+
+- 🔴 If Meilisearch is secured, accessing this route without the `Authorization` header returns a [missing_authorization_header](0061-error-format-and-definitions.md#missing_authorization_header) error.
+- 🔴 If Meilisearch is secured, accessing this route with a key that does not have permissions (i.e. other than the master-key) returns an [invalid_api_key](0061-error-format-and-definitions.md#invalid_api_key) error.
+
+---
+
+### 3.3. CLI Definition
```
Comment (gmourier) [inline text/0258-snapshots-api.md:55]: I'm not sure we have to display the full details on that file @irevoire. Those are already described in the instances-options one, maybe a link to that file is enough.

This way, we will have only one place to modify if needed in the future

**Comment by macraig** [comment]
@irevoire can you make the PR point to `release-v1.5.0` instead of `main`?

**Review by gmourier — APPROVED** [review]
🚜

**Pull request merged**